Source/CarthageKit/Project.swift

```diff
@@ -295,6 +299,16 @@ public final class Project {
 			}
 			.startOnQueue(cachedVersionsQueue)
 	}
+	
+	/// Loads the Cartfile for the given dependency, at the given version.
+	private func cartfileForDependency(dependency: Dependency<PinnedVersion>) -> SignalProducer<Cartfile, CarthageError> {
+		let revision = dependency.version.commitish
+		return self.cloneOrFetchDependency(dependency.project, commitish: revision).flatMap(.Concat) { repositoryURL in
+			return contentsOfFileInRepository(repositoryURL, CarthageProjectCartfilePath, revision: revision)
+				.flatMapError { _ in .empty }
+				.attemptMap(Cartfile.fromString)
+		}
```

Comment:
We try to keep signal chains as flat as possible to increase readability.

```swift
return self.cloneOrFetchDependency(dependency.project, commitish: revision)
	.flatMap(.Concat) { repositoryURL in
		return contentsOfFileInRepository(repositoryURL, CarthageProjectCartfilePath, revision: revision)
	}
	.flatMapError { _ in .empty }
	.attemptMap(Cartfile.fromString)
```

Would you mind reformatting this accordingly?